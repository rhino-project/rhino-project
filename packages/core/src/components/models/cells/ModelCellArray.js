import { useCallback } from 'react';
import { useGlobalComponent } from 'rhino/hooks/overrides';
import CellString from 'rhino/components/table/cells/CellString';

export const ModelCellArrayBase = ({ getValue, ...props }) => {
  const syntheticGetValue = useCallback(() => getValue()?.join(', '), [
    getValue
  ]);

  return <CellString getValue={syntheticGetValue} {...props} />;
};

const ModelCellArray = (props) =>
  useGlobalComponent('ModelCellArray', ModelCellArrayBase, props);

export default ModelCellArray;
