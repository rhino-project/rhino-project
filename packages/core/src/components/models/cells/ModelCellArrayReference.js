import { useCallback } from 'react';
import { useGlobalComponentForAttribute } from '../../../hooks/overrides';
import { CellString } from '../../table/cells/CellString';
import { isFunction } from 'lodash-es';

export const ModelCellArrayReferenceBase = ({
  getValue,
  accessor = 'display_name',
  ...props
}) => {
  const syntheticGetValue = useCallback(
    () =>
      getValue()
        ?.map((v) =>
          isFunction(accessor) ? accessor(getValue()) : v[accessor]
        )
        ?.join(', '),
    [accessor, getValue]
  );

  return <CellString getValue={syntheticGetValue} {...props} />;
};

export const ModelCellArrayReference = (props) =>
  useGlobalComponentForAttribute(
    'ModelCellArrayReference',
    ModelCellArrayReferenceBase,
    props
  );
