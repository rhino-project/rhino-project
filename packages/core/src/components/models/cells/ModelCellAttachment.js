import { useCallback, useMemo } from 'react';
import { useGlobalOverrides } from 'rhino/hooks/overrides';
import CellLink from 'rhino/components/table/cells/CellLink';

export const ModelCellAttachmentBase = ({ children, getValue, ...props }) => {
  const syntheticGetValue = useCallback(() => getValue()?.url, [getValue]);
  const linkText = useMemo(() => children || getValue()?.display_name, [
    children,
    getValue
  ]);

  return (
    <CellLink getValue={syntheticGetValue} {...props}>
      {linkText}
    </CellLink>
  );
};

const defaultComponents = { ModelCellAttachment: ModelCellAttachmentBase };

const ModelCellAttachment = ({ overrides, ...props }) => {
  const { ModelCellAttachment } = useGlobalOverrides(
    defaultComponents,
    overrides,
    props
  );

  return <ModelCellAttachment {...props} />;
};

export default ModelCellAttachment;
