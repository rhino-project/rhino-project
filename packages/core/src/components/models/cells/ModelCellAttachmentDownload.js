import { useCallback, useMemo } from 'react';
import { useGlobalComponentForAttribute } from '../../../hooks/overrides';
import { CellLink } from '../../table/cells/CellLink';

export const ModelCellAttachmentDownloadBase = ({
  children,
  getValue,
  ...props
}) => {
  const syntheticGetValue = useCallback(
    () => getValue()?.url_attachment,
    [getValue]
  );
  const linkText = useMemo(
    () => children || getValue()?.display_name,
    [children, getValue]
  );

  return (
    <CellLink getValue={syntheticGetValue} {...props}>
      {linkText}
    </CellLink>
  );
};

export const ModelCellAttachmentDownload = (props) =>
  useGlobalComponentForAttribute(
    'ModelCellAttachmentDownload',
    ModelCellAttachmentDownloadBase,
    props
  );
