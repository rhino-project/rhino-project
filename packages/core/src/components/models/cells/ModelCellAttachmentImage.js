import { useCallback, useMemo } from 'react';
import { useGlobalComponent } from 'rhino/hooks/overrides';
import CellImage from 'rhino/components/table/cells/CellImage';

export const ModelCellAttachmentImageBase = ({ getValue, ...props }) => {
  const syntheticGetValue = useCallback(() => getValue()?.url, [getValue]);

  const altText = useMemo(() => getValue()?.display_name || getValue()?.url, [
    getValue
  ]);

  return <CellImage alt={altText} getValue={syntheticGetValue} {...props} />;
};

const ModelCellAttachmentImage = (props) =>
  useGlobalComponent(
    'ModelCellAttachmentImage',
    ModelCellAttachmentImageBase,
    props
  );

export default ModelCellAttachmentImage;
