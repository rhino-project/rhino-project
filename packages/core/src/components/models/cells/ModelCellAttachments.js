import { useCallback } from 'react';
import { useGlobalComponent } from 'rhino/hooks/overrides';
import CellString from 'rhino/components/table/cells/CellString';

export const ModelCellAttachmentsBase = ({ getValue, ...props }) => {
  const syntheticGetValue = useCallback(
    () => (getValue()?.length ? `${getValue()?.length} files` : undefined),
    [getValue]
  );

  return <CellString getValue={syntheticGetValue} {...props} />;
};

const ModelCellAttachments = (props) =>
  useGlobalComponent('ModelCellAttachments', ModelCellAttachmentsBase, props);

export default ModelCellAttachments;
