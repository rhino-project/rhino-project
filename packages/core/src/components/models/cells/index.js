export * from './ModelCellArray';
export * from './ModelCellArrayReference';
export * from './ModelCellAttachment';
export * from './ModelCellAttachmentDownload';
export * from './ModelCellAttachmentImage';
export * from './ModelCellAttachments';
export * from './ModelCellBoolean';
export * from './ModelCellBooleanIcon';
export * from './ModelCellCountry';
export * from './ModelCellCurrency';
export * from './ModelCellDate';
export * from './ModelCellDateTime';
export * from './ModelCellDateTimeDistance';
export * from './ModelCellEnum';
export * from './ModelCellFloat';
export * from './ModelCellInteger';
export * from './ModelCellLinkEmail';
export * from './ModelCellLinkTelephone';
export * from './ModelCellReference';
export * from './ModelCellString';
export * from './ModelCellTime';
export * from './ModelEditableCellBoolean';
export * from './ModelEditableCellReference';
export * from './ModellCellIdentifier';
