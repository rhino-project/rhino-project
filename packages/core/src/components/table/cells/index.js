export * from './CellBadge';
export * from './CellBoolean';
export * from './CellBooleanIcon';
export * from './CellCountry';
export * from './CellCurrency';
export * from './CellDate';
export * from './CellDateTime';
export * from './CellDateTimeDistance';
export * from './CellDateTimeRelative';
export * from './CellFloat';
export * from './CellImage';
export * from './CellInteger';
export * from './CellLink';
export * from './CellLinkEmail';
export * from './CellLinkTelephone';
export * from './CellReference';
export * from './CellString';
export * from './CellTime';
