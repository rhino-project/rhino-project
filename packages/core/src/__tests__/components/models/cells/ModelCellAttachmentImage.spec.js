import { render } from '@testing-library/react';
import { ModelCellAttachmentImage } from '../../../../components/models/cells/ModelCellAttachmentImage';
import { sharedCellTests } from './sharedCellTests';

describe('ModelCellAttachmentImage', () => {
  const getValue = () => ({
    url: 'https://example.com/image.png'
  });

  sharedCellTests(ModelCellAttachmentImage);

  it(`uses url for alt as fallback`, () => {
    const { asFragment } = render(
      <ModelCellAttachmentImage
        model={{
          properties: {
            dummy: {
              type: 'reference'
            }
          }
        }}
        path="dummy"
        getValue={getValue}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it(`uses passed in alt`, () => {
    const { asFragment } = render(
      <ModelCellAttachmentImage
        model={{
          properties: {
            dummy: {
              type: 'reference'
            }
          }
        }}
        alt="foo"
        path="dummy"
        getValue={getValue}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
