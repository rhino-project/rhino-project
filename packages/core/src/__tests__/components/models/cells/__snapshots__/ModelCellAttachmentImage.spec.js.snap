// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelCellAttachmentImage should render the component with className from inherited props 1`] = `
<DocumentFragment>
  <div
    class="dummy-class"
  >
    baz
  </div>
</DocumentFragment>
`;

exports[`ModelCellAttachmentImage should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <div>
    -
  </div>
</DocumentFragment>
`;

exports[`ModelCellAttachmentImage should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <div>
    baz
  </div>
</DocumentFragment>
`;

exports[`ModelCellAttachmentImage should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellAttachmentImage should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellAttachmentImage should render with global override shorthand for model and attribute 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCellAttachmentImage uses passed in alt 1`] = `
<DocumentFragment>
  <img
    alt="foo"
    height="50"
    maxwidth="100%"
    src="https://example.com/image.png"
  />
</DocumentFragment>
`;

exports[`ModelCellAttachmentImage uses url for alt as fallback 1`] = `
<DocumentFragment>
  <img
    alt="https://example.com/image.png"
    height="50"
    maxwidth="100%"
    src="https://example.com/image.png"
  />
</DocumentFragment>
`;
